### PR TITLE
Note function signature param/result features for validation

### PR DESCRIPTION
**Pull request by kripken**
As with #5535, this was not noticed because it can only happen on very
small modules where the param/result type appears nowhere else but
in a function signature, but my wip fuzzer branch now finds it.

### CODE REVIEW

File: test/lit/validation/ref-func-simd.wast [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  (type $simd-user (func (param v128)))
+
+  (global $g (ref null $simd-user) (ref.null func))
+)
+
+;; But it passes with the feature enabled.
+;; RUN: wasm-opt --enable-reference-types --enable-simd %s
+
+
+
```
Comment (tlively) [inline test/lit/validation/ref-func-simd.wast:16]: It would be good to remove the unnecessary whitespace.

**Review by tlively — APPROVED** [review]

**Comment by tlively** [comment]
We have a `HeapTypeGraphWalker` utility in wasm-type.cpp that might be useful.

**Comment by kripken** [comment]
Thanks, but I think we just need a recursion guard? I pushed that now. Diff without whitespace is much smaller, this just wraps the inside in a function.

**Review by tlively — COMMENTED** [review]
It seems fishy that we recurse into function types but not structs or arrays.

**Comment by kripken** [comment]
Good point, that was wrong too. I pushed a fix now that `HeapTypeGraphWalker`, it does seem like we should not duplicate walking logic here.

I also reused `getTypeChildren` from your PR #5541 so this should land after that.
